src/single-spa-alpinejs.js

```
@@ -98,6 +154,9 @@ function unmount(opts, props) {
 
     domElement.innerHTML = "";
     // TODO, remove any global functions that were used for x-init
+    if (window.hasOwnProperty("xInitFn")) {
```

We want single-spa-alpinejs' unmount function to die with a loud error if it cannot properly unmount. This is better than silent failure, since the user is aware of it and can fix it. The if statement on this line creates the possible of silent failures.

```suggestion
```

---

Added a check for x-init
```
 // remove any global functions that were used for x-init
    if (opts.xInit) {
      if (
        !window.hasOwnProperty("xInitFn") ||
        typeof window.xInitFn[`${appName}`] === "undefined"
      ) {
        throw new Error(
          `single-spa-alpinejs: global function for xInit not found. Optional parameter opts.xInit if provided must be as a function that returns a promise`
        );
      }
      delete window.xInitFn[`${appName}`];
    }
```